lib/dialyzer/src/dialyzer_plt.erl

```diff
@@ -531,17 +531,19 @@ compute_md5_from_files(Files) ->
   lists:keysort(1, [{F, compute_md5_from_file(F)} || F <- Files]).
 
 compute_md5_from_file(File) ->
-  case filelib:is_regular(File) of
-    false ->
+  case beam_lib:all_chunks(File) of
+    {ok, _, Chunks} ->
+      %% We cannot use beam_lib:md5 because it does not consider
+      %% the debug_info chunk, where typespecs are likely stored.
+      %% So we consider almost all chunks except the useless ones.
+      Filtered = [[ID, Chunk] || {ID, Chunk} <- Chunks, ID /= "CInf", ID /= "Docs"],
```

Comment:
I suggest to use exact comparison operators here.

Reply:
Keep two commits, please.